src/main/java/kafka/KafkaClientFactory.java

```diff
@@ -50,13 +50,16 @@ public static ReceiverOptions<String, String> createReceiverOptions() {
         var props = getAuthProperties();
         props.put(ConsumerConfig.GROUP_ID_CONFIG, Config.GROUP_ID);
         props.put(
-            ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,
-            "org.apache.kafka.common.serialization.StringDeserializer"
+                ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,
+                "org.apache.kafka.common.serialization.StringDeserializer"
         );
         props.put(
-            ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
-            "org.apache.kafka.common.serialization.StringDeserializer"
+                ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
+                "org.apache.kafka.common.serialization.StringDeserializer"
         );
+        if(Config.ASSIGNMENT_STRATEGY != null && Config.ASSIGNMENT_STRATEGY.size()>0){
+            props.put(ConsumerConfig.PARTITION_ASSIGNMENT_STRATEGY_CONFIG, String.join(",",Config.ASSIGNMENT_STRATEGY));
```

Comment:
Why is it a list?

Reply:
The value for the config is a list of strategies ordered by preference.
Meaning whatever is supported by the client/broker first will be used, if not it will try to use the next... remember this?
https://github.com/osskit/dafka-consumer/blob/e32f8a1720abd8e13d6ec38e0ecfd9105fd07397/src/main/java/kafka/KafkaClientFactory.java#L66